src/Annotations/Serializer/CommonMark/functions.php

```diff
@@ -2,9 +2,25 @@
 
 namespace eLife\Annotations\Serializer\CommonMark;
 
-function clean_paragraph($text)
+use League\CommonMark\Util\Xml;
+
+function clean_paragraph(string $text) : string
 {
     $allowed_tags = '<i><sub><sup><span><del><math><a><br><caption>';
 
     return strip_tags($text, $allowed_tags);
 }
+
+function escape_math(string $text) : string
+{
+    // Escape MathML.
+    $escaped = preg_replace_callback('~(?P<before><math[^>]*>)(?P<mathml>.*)(?P<after></math>)~s', function ($match) {
+        return Xml::escape($match['before'].$match['mathml'].$match['after']);
+    }, $text);
+    // Escape LaTeX.
+    $escaped = preg_replace_callback('~(?P<before>\$\$)(?P<latex>.+)(?P<after>\$\$)~s', function ($match) {
+        return Xml::escape($match['before'].$match['latex'].$match['after']);
+    }, $escaped);
+
+    return $escaped;
+}
```

Comment:
also, I'm not sure whether an "unescaping" is necessary

Reply:
unescaping is not necessary as LaTeX and MathML are being presented as escaped in the api response.